src/controllers/teamCtrl.js

```diff
@@ -44,82 +48,129 @@ exports.updateTeamName = (req, res) => {
 };
 
 exports.addUserToTeam = (req, res) => {
-  Team.findByIdAndUpdate(
-    req.params.teamId,
-    {
-      $push: {
-        users: {
-          id: req.body.userId,
-          role: req.body.role,
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if user not in the team
+  Team.findById(req.params.teamId)
```

Comment:
Same as before

Reply:
Same, to check if user was not already in team

Reply:
The other conversations can be resolved ?